README.adoc

```diff
@@ -302,6 +304,53 @@ If you are using VisualStudio Code, you can install the following `launcher.json
 }
 ----
 
+==== GUI Frontend
+
+If you are using VisualStudio Code, you can install the following `launcher.json` that is then used to launch Chrome to debug the Kiali Server GUI frontend in the debugger. Run the `make -e YARN_START_URL=http://localhost:20001/kiali yarn-start` first to start the GUI frontend locally. Make sure your YARN_START_URL is valid - it should point to a running Kiali Server. In this example, it assumes you ran a port-forward so that the Kiali Server can be reached at localhost:20001. If running in minikube, you could alternatively use `$(minikube ip):80` as the host/port assuming an Ingress is set up to access Kiali.
+
+[source,json]
+----
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Launch Chrome",
+            "type": "chrome",
+            "request": "launch",
+            "url": "http://localhost:3001",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
+----
+
+=== Debugging With Google Chrome Developer Tools
+
+You can debug the Kiali GUI directly inside of Google Chrome using the Chrome Developer Tools.
+
+First, run the Kiali Server backend, either normally within a cluster or via `run-server.sh` as explained earlier. Determine what the Kiali URL is before moving to the next step. For example, if you run Kiali in minikube, and you set up a port-forward that exposes it, the URL to remember will be `http://localhost:20001/kiali`.
+
+Second, run the GUI frontend using make: `make -e YARN_START_URL=http://<kiali-url> yarn-start` where `<kiali-url>` is the URL you determined from the previous step. For example, if the Kiali Server is running in minikube with a port-forward exposing it, then run `make -e YARN_START_URL=http://localhost:20001/kiali yarn-start`. This command will start the Kiali GUI frontend on a local endpoint. After the `yarn-start` command is ready, look at the output for the "Local" URL you use to access it. The output will look something like this:
```

Comment:
@ferhoyos I have two questions/comments:

1. Why do we need to say PORT=3001 here? What uses that env var? The default Grafana port that run-kiali.sh will use to port-forward is 3000 (not 3001) -- see: https://github.com/kiali/kiali/blob/v1.64.0/hack/run-kiali.sh#L84
This is why I suggest using 3001 in the URL. I think 3001 will not conflict with anything by default. So I don't know why we need "PORT" - I don't know what uses that env var.

2. As for this:

> It does not matter if the Kiali server is running on minikube or openshift

The reason I say "if on minikube..." is because the default Kiali root path is different between OpenShift and non-OpenShift clusters (see the [code here](https://github.com/kiali/kiali-operator/blob/v1.64.0/roles/default/kiali-deploy/tasks/main.yml#L267-L271)). On minikube, the default root path used by kiali is `/kiali` but on OpenShift the default root path is `/`... so my example is specifically for minikube (I don't think it will work on OpenShift because there is no root path `/kiali` on OpenShift).

Reply:
OK, I think I get it now. Port 3000 is what yarn wants to use, and if you have Grafana set up, that's the conflict. We need to pass "PORT" because yarn will take that and use it. So saying PORT=3001 will move yarn away from the grafana port.

Reply:
I added some more around this whole 3000 vs 3001.